src/chainstate/stacks/db/mod.rs

```diff
@@ -752,14 +773,152 @@ impl StacksChainState {
+                                amount: balance.amount.into(),
+                            }),
+                        );
+                        allocation_events.push(mint_event);
+                    }
+                });
+                info!("Done initializing chain with {} balances", balances_count);
             }
+
+            let allocations_tx = StacksTransaction::new(
+                tx_version.clone(),
+                boot_code_auth.clone(),
+                TransactionPayload::TokenTransfer(
+                    PrincipalData::Standard(boot_code_address.into()),
+                    0,
+                    TokenTransferMemo([0u8; 34]),
+                ),
+            );
+            let allocations_receipt = StacksTransactionReceipt::from_stx_transfer(
+                allocations_tx,
+                allocation_events,
+                Value::okay_true(),
+                ExecutionCost::zero(),
+            );
+            receipts.push(allocations_receipt);
+
+            let mut total_vesting_amount: u128 = 0;
+            if let Some(get_schedules) = boot_data.get_bulk_initial_vesting_schedules.take() {
+                info!("Initializing chain with vesting schedules");
+                let mut vesting_schedule_count = 0;
+                clarity_tx.connection().as_transaction(|clarity| {
+                    let initial_vesting_schedules = get_schedules();
+                    let mut unlocks_per_blocks: HashMap<u64, u32> = HashMap::new();
+                    let lockup_contract_id = boot_code_id("lockup");
+                    for schedule in initial_vesting_schedules {
+                        total_vesting_amount = total_vesting_amount
+                            .checked_add(schedule.amount as u128)
+                            .expect("FATAL: vesting STX overflow");
+                        vesting_schedule_count = vesting_schedule_count + 1;
+                        let value = unlocks_per_blocks.get(&schedule.block_height).unwrap_or(&0);
+                        let index = value + 1;
+                        unlocks_per_blocks.insert(schedule.block_height, index);
+
+                        clarity
+                            .with_clarity_db(|db| {
+                                let key = TupleData::from_data(vec![
+                                    (
+                                        "stx-height".into(),
+                                        Value::UInt(schedule.block_height.into()),
+                                    ),
+                                    ("index".into(), Value::UInt(index.into())),
+                                ])
+                                .unwrap();
+
+                                let value = TupleData::from_data(vec![
+                                    ("owner".into(), Value::Principal(schedule.address.into())),
+                                    ("metadata".into(), Value::buff_from_byte(0)),
```

Comment:
If we're not going to pass on the `metadata` from Stacks 1.0, we can drop it from the `lockup` contract altogether.

Reply:
👍 
@jcnelson just to be clear, this `metadata` was originally to match the metadata column in the Stacks 1.0 `accounts` table with addresses that had a `lock_id` in the future. However, these have since all unlocked (since block 657212). The remaining vesting entries are from the `account_vesting` table, which does not contain the equivalent `metadata`. 
Does that make sense?

Reply:
Yup, makes sense.  It's fine to leave as-is as well -- up to you.

Reply:
Looks like @lgalabru has taken care of this in follow-up PR https://github.com/blockstack/stacks-blockchain/pull/2109
(we should try to get this one merged to unblock that one)